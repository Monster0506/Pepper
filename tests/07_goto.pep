%% SIMPLE GOTO EXAMPLE
SHOW("Before jump")
GOTO 6;
SHOW("This will be skipped")
SHOW("This will be skipped too")
SHOW("After jump")

%% CONDITIONAL GOTO
LET x: int = 5
REAS x = x 1 +
SHOW(x)
GOTO 10; x < 10
SHOW("x < 10 : TRUE")
GOTO 10; x < 16


%% NESTED LOOPS 
LET a: int = 0
LET b: int = -1
REAS a = 1 a +
REAS b = 1 b +
SHOW("Inner:  " + b + "           Outer: " + a )
GOTO 21; b < 20
REAS b = -1
GOTO 20; a < 5


%% GOTO with labels
LET m: int = 0
LBL a;
REAS m = m 1 +
SHOW(m)
GOTO a; m < 15
