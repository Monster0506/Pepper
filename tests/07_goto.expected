Before jump
After jump
6
7
8
9
10
x < 10 : TRUE
11
x < 10 : TRUE
12
x < 10 : TRUE
13
x < 10 : TRUE
14
x < 10 : TRUE
15
x < 10 : TRUE
16
x < 10 : TRUE
Inner:  0           Outer: 1
Inner:  1           Outer: 1
Inner:  2           Outer: 1
Inner:  3           Outer: 1
Inner:  4           Outer: 1
Inner:  5           Outer: 1
Inner:  6           Outer: 1
Inner:  7           Outer: 1
Inner:  8           Outer: 1
Inner:  9           Outer: 1
Inner:  10           Outer: 1
Inner:  11           Outer: 1
Inner:  12           Outer: 1
Inner:  13           Outer: 1
Inner:  14           Outer: 1
Inner:  15           Outer: 1
Inner:  16           Outer: 1
Inner:  17           Outer: 1
Inner:  18           Outer: 1
Inner:  19           Outer: 1
Inner:  20           Outer: 1
Inner:  0           Outer: 2
Inner:  1           Outer: 2
Inner:  2           Outer: 2
Inner:  3           Outer: 2
Inner:  4           Outer: 2
Inner:  5           Outer: 2
Inner:  6           Outer: 2
Inner:  7           Outer: 2
Inner:  8           Outer: 2
Inner:  9           Outer: 2
Inner:  10           Outer: 2
Inner:  11           Outer: 2
Inner:  12           Outer: 2
Inner:  13           Outer: 2
Inner:  14           Outer: 2
Inner:  15           Outer: 2
Inner:  16           Outer: 2
Inner:  17           Outer: 2
Inner:  18           Outer: 2
Inner:  19           Outer: 2
Inner:  20           Outer: 2
Inner:  0           Outer: 3
Inner:  1           Outer: 3
Inner:  2           Outer: 3
Inner:  3           Outer: 3
Inner:  4           Outer: 3
Inner:  5           Outer: 3
Inner:  6           Outer: 3
Inner:  7           Outer: 3
Inner:  8           Outer: 3
Inner:  9           Outer: 3
Inner:  10           Outer: 3
Inner:  11           Outer: 3
Inner:  12           Outer: 3
Inner:  13           Outer: 3
Inner:  14           Outer: 3
Inner:  15           Outer: 3
Inner:  16           Outer: 3
Inner:  17           Outer: 3
Inner:  18           Outer: 3
Inner:  19           Outer: 3
Inner:  20           Outer: 3
Inner:  0           Outer: 4
Inner:  1           Outer: 4
Inner:  2           Outer: 4
Inner:  3           Outer: 4
Inner:  4           Outer: 4
Inner:  5           Outer: 4
Inner:  6           Outer: 4
Inner:  7           Outer: 4
Inner:  8           Outer: 4
Inner:  9           Outer: 4
Inner:  10           Outer: 4
Inner:  11           Outer: 4
Inner:  12           Outer: 4
Inner:  13           Outer: 4
Inner:  14           Outer: 4
Inner:  15           Outer: 4
Inner:  16           Outer: 4
Inner:  17           Outer: 4
Inner:  18           Outer: 4
Inner:  19           Outer: 4
Inner:  20           Outer: 4
Inner:  0           Outer: 5
Inner:  1           Outer: 5
Inner:  2           Outer: 5
Inner:  3           Outer: 5
Inner:  4           Outer: 5
Inner:  5           Outer: 5
Inner:  6           Outer: 5
Inner:  7           Outer: 5
Inner:  8           Outer: 5
Inner:  9           Outer: 5
Inner:  10           Outer: 5
Inner:  11           Outer: 5
Inner:  12           Outer: 5
Inner:  13           Outer: 5
Inner:  14           Outer: 5
Inner:  15           Outer: 5
Inner:  16           Outer: 5
Inner:  17           Outer: 5
Inner:  18           Outer: 5
Inner:  19           Outer: 5
Inner:  20           Outer: 5
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15